### PR TITLE
fix: pause video & audio elements along with replay playback

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

Video (and audio) elements in RRWeb playback are not paused when the player is.

## Changes

- Find playing video elements
- Pause them within the `setPause` function
- Store the in the cache
- Play them again once the playback starts again

## How did you test this code?

Manually

https://github.com/PostHog/posthog/assets/6685876/bc42e475-3347-4802-a4cf-2b30e932b471

(The levels of inception in that video 🤯)

### CODE REVIEW

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts [hunk 1]
```diff
@@ -818,6 +825,23 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
...
             }
         },
+        pauseIframePlayback: () => {
+            const iframe = values.rootFrame?.querySelector('iframe')
+            const iframeDocument = iframe?.contentWindow?.document
+            if (!iframeDocument) {
+                return
+            }
+
+            const mediaElements: HTMLMediaElement[] = Array.from(iframeDocument.getElementsByTagName('audio'))
```
Comment (benjackwhite) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:835]: PR says video elements but I only see audio element 🤔 ??

**Review by benjackwhite — COMMENTED** [review]
Code wise looks good. Haven't tested it myself though. Would be cool if we could add some video / audio elements to the playground nextjs app like we do with animations so that its easy to validate this in the future
([see here](https://github.com/PostHog/posthog-js/blob/master/playground/nextjs/pages/animations.tsx))

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts (hunk 1 above)
Comment (benjackwhite) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:835]: One other thing (can be a different PR) - I think it might be a good idea to allow audio in general to be disabled. Especially given people watch recordings at faster / speeds or with a UX focus, the audio might not be that helpful...

**Comment by daibhin** [comment]
@benjackwhite added the examples in https://github.com/PostHog/posthog-js/pull/771. Fixed up the implementation and should be good for a review again

**Review by benjackwhite — APPROVED** [review]
Loooooks good!

**Comment by patricsteiner** [comment]
thanks @daibhin for fixing this so quickly :)
